### DISCUSSION TITLE
withAuth middleware without matchers is incompatible with custom pages

### DISCUSSION THREAD

**Discussion by ChristianIvicevic** [nextauthjs/next-auth#4984] [Help]
I tried out the `withAuth` middleware:
```typescript
import { withAuth } from 'next-auth/middleware';
export default withAuth({});
```
Since it has no `config` and thus no `matchers` it basically applies auth protection to the entire application. Combining that with a custom login page I am running into an infinite loop of redirects:
```typescript
import { PrismaAdapter } from '@next-auth/prisma-adapter';
import NextAuth from 'next-auth';
import DiscordProvider from 'next-auth/providers/discord';
import { prisma } from '../../../server/db/client';

export default NextAuth({
  adapter: PrismaAdapter(prisma),
  providers: [
    DiscordProvider({
      clientId: env.DISCORD_CLIENT_ID,
      clientSecret: env.DISCORD_CLIENT_SECRET,
    }),
  ],
  session: {
    strategy: 'jwt',
  },
  pages: {
    signIn: '/auth/login',
  },
});
```
I can only assume that the middleware attempts to apply to the `/auth/login` route as well and redirects over and over again with an ever increasing `callbackUrl`. What options are there to sort that out accordingly?

I'd already be happy by either skipping the `/auth` pages in the middleware or apply the middleware to everything BUT `/` and make the index have a custom login using `signIn('discord');`.

**Comment by jacobhjkim** [comment]
I am having the same issue.
- `next-auth`: `4.10.2`
- `next`: `12.2.3`

**Comment by SeeNoMore** [comment]
I have same issue.

- `next-auth`: `4.24.3`
- `next`: `13.5.6`
